### PR TITLE
Adding Contribution guidelines 

### LINKED ISSUE #183: Missing CONTRIBUTING.md file
All K8s subrepositories should have a CONTRIBUTING.md file, which at the minimum should point to https://github.com/kubernetes/community/blob/master/contributors/guide/README.md. Care should be taken that all information is in sync with the contributor guide.

Subrepositories may also have contributing guidelines specific to that repository. They should be explicitly documented and explained in the CONTRIBUTING.md

Ref: kubernetes/community#1832

### PR DESCRIPTION
**Pull request by RA489**
Adding Contribution guidelines for kubernetes federation-v2.
fixes #183

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+### Learn a bit about the Kubernetes Federation v2
+To learn more about the Kubernetes Federation v2 project, please read through the User Guide:
+https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/userguide.md
+
+If you are interested in contributing to Kubernetes Federation v2 project, please read through the Development Guide:
+https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/development.md
+
+
+### Contributing A Patch
```
Comment (font) [inline CONTRIBUTING.md]: How about changing the title to "Submitting a Pull Request (PR)"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+1. Submit an issue by describing your proposed change to the repo in question.
+2. The repo owners will respond to your issue rapidly.
+3. Fork the repo make your changes and test it.
```
Comment (font) [inline CONTRIBUTING.md]: s/repo/repo,/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Contributing Guidelines
+This repo contains an in-progress prototype of some of the foundational aspects of V2 of Kubernetes Federation. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
+
+### Sign the CLA
+Kubernetes projects require that you sign a Contributor License Agreement (CLA) before we can accept your pull requests. Please see https://github.com/kubernetes/community/blob/master/CLA.md for more information
+
+### Learn a bit about the Kubernetes Federation v2
```
Comment (font) [inline CONTRIBUTING.md]: I would retitle this as "Learn About Kubernetes Federation v2" and I would also add a link to the main README where it talks about the project and concepts https://github.com/kubernetes-sigs/federation-v2#kubernetes-federation-v2.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+To learn more about the Kubernetes Federation v2 project, please read through the User Guide:
+https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/userguide.md
+
+If you are interested in contributing to Kubernetes Federation v2 project, please read through the Development Guide:
+https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/development.md
+
+
+### Contributing A Patch
+1. Submit an issue by describing your proposed change to the repo in question.
+2. The repo owners will respond to your issue rapidly.
```
Comment (font) [inline CONTRIBUTING.md]: We should add a step about signing the CLA.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Contributing Guidelines
+This repo contains an in-progress prototype of some of the foundational aspects of V2 of Kubernetes Federation. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
```
Comment (font) [inline CONTRIBUTING.md]: Replace in-progress prototype to say alpha release since we've officially released our first alpha.

**Comment by RA489** [comment]
@font I have updated as per your comments.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# Contributing Guidelines
+This repo contains alpha release of some of the foundational aspects of V2 of Kubernetes Federation. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
```
Comment (font) [inline CONTRIBUTING.md]: s/alpha/an alpha/

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+To learn more about the Kubernetes Federation v2 project, please read through the User Guide:
+https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/userguide.md
+
+If you are interested in contributing to Kubernetes Federation v2 project, please read through the Development Guide:
+https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/development.md
+
+
+### Submitting a Pull Request (PR)
+1. It is require that you sign a Contributor License Agreement (CLA) before we can accept your pull requests.Please see            https://github.com/kubernetes/community/blob/master/CLA.md for more information
```
Comment (font) [inline CONTRIBUTING.md]: Just a generic "Sign the CLA" step is all that is needed. Or you can reference the section above in this guide where it talks about signing the CLA.  Let's not duplicate the same content.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# Contributing Guidelines
+This repo contains alpha release of some of the foundational aspects of V2 of Kubernetes Federation. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
+
+### Sign the CLA
+Kubernetes projects require that you sign a Contributor License Agreement (CLA) before we can accept your pull requests. Please see https://github.com/kubernetes/community/blob/master/CLA.md for more information
+
+### Learn About Kubernetes Federation v2
+If you want to learn about Kubernetes Federation v2,please read through the doc:
```
Comment (font) [inline CONTRIBUTING.md]: Space after the comma.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+# Contributing Guidelines
+This repo contains alpha release of some of the foundational aspects of V2 of Kubernetes Federation. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
+
+### Sign the CLA
```
Comment (font) [inline CONTRIBUTING.md]: You started off with just 1 `#` for the heading size and then went to 3. I would use only 2 `#` for the heading size for all these sub-sections.

**Comment by RA489** [comment]
@font I have updated as per your previous comments.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+This repo contains an alpha release of some of the foundational aspects of V2 of Kubernetes Federation. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
+
+## Sign the CLA
+Kubernetes projects require that you sign a Contributor License Agreement (CLA) before we can accept your pull requests. Please see https://github.com/kubernetes/community/blob/master/CLA.md for more information
+
+## Learn About Kubernetes Federation v2
+If you want to learn about Kubernetes Federation v2, please read through the doc:
+https://github.com/kubernetes-sigs/federation-v2#kubernetes-federation-v2
+
+To learn more about the Kubernetes Federation v2 project, please read through the User Guide:
```
Comment (font) [inline CONTRIBUTING.md]: The user guide is really a guide on using the Kubernetes Federation v2 project. So I would rephrase to something like "To learn how to use Kubernetes Federation v2, please read...." or "To deploy Kubernetes Federation v2, ..." or similar.

**Comment by RA489** [comment]
@font I have updated your last comment!
